util/opentelemetry-util-genai/src/opentelemetry/util/genai/metrics.py

```diff
@@ -64,23 +78,21 @@ def record(
             attributes[GenAI.GEN_AI_RESPONSE_MODEL] = (
                 invocation.response_model_name
             )
-
-        # Calculate duration from span timing or invocation monotonic start
-        duration_seconds: Optional[float] = None
-        if invocation.monotonic_start_s is not None:
-            duration_seconds = max(
-                timeit.default_timer() - invocation.monotonic_start_s, 0.0
+        if invocation.server_address:
+            attributes[ServerAttributes.SERVER_ADDRESS] = (
+                invocation.server_address
             )
+        if invocation.server_port is not None:
+            attributes[ServerAttributes.SERVER_PORT] = invocation.server_port
+        if invocation.metric_attributes:
+            attributes.update(invocation.metric_attributes)
 
         span_context = set_span_in_context(span)
         if error_type:
-            attributes["error.type"] = error_type
+            attributes[ErrorAttributes.ERROR_TYPE] = error_type
 
-        if (
-            duration_seconds is not None
-            and isinstance(duration_seconds, Number)
```

Comment:
it can't really be anything else (if not None)
and there is no reason not to record 0 duration - why would we drop points for it? Assuming it can happen, we'd be skewing counts and everything else